sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/data/BeamFnDataOutboundObserver.java

```diff
@@ -73,7 +73,11 @@ public void close() throws Exception {
   }
 
   @Override
-  public void flush() throws IOException {}
+  public void flush() throws IOException {
```

Comment:
You could make periodicFlush throw IOException and invoke it instead. You would move the try/catch logic into a lambda where we schedule the thread in the constructor.

It would also make sense to rename periodicFlush() to flush() and what is currently named flush() to flushInternal() since we don't want to expose the flushLock if we don't have to.